test/DynamoCoreTests/UserDataMigrationTests.cs

```diff
@@ -357,5 +359,39 @@ public void TestMigration()
             Assert.AreEqual(currentVersionDir,
                 targetMigrator.PreferenceSettings.CustomPackageFolders[0]);
         }
+
+        [Test]
+        [Category("UnitTests")]
+        public void TestSelectedDownloadPathIsNotMigrated()
+        {
+            // Create 1.3, and 2.0 version user data directories in Temp folder
+            string userDataDir;
+            CreateMockDirectoriesAndFiles(out userDataDir);
+
+            var sourceVersionDir = Path.Combine(userDataDir, "1.3");
+            var settingsFilePath = Path.Combine(sourceVersionDir, "DynamoSettings.xml");
+
+            // Create PreferenceSettings.xml file in 1.3
+            CreateMockPreferenceSettingsFile(settingsFilePath, sourceVersionDir);
+
+            // Create mock objects for IPathManager and IPathResolver
+            var mockPathManager = new Mock<IPathManager>();
+
+            var currentVersionDir = Path.Combine(userDataDir, "2.0");
+
+            mockPathManager.Setup(x => x.UserDataDirectory).Returns(() => currentVersionDir);
+            
+
+            // Test MigrateBetweenDynamoVersions
+            var targetMigrator = DynamoMigratorBase.MigrateBetweenDynamoVersions(
+                mockPathManager.Object);
+
+            var sourcePrefs = PreferenceSettings.Load(settingsFilePath);
+            Assert.AreEqual(sourceVersionDir, sourcePrefs.SelectedPackagePathForInstall);
+
+            // Assert that new SelectedPackagePath is not equal to the old path.
+            Assert.AreNotEqual(sourcePrefs.SelectedPackagePathForInstall,
+                targetMigrator.PreferenceSettings.SelectedPackagePathForInstall);
+        }
```

Comment:
Can we assert that `targetMigrator.PreferenceSettings.SelectedPackagePathForInstall` is equal to `currentVersionDir`?

Reply:
@aparajit-pratap I couldn't get this working because in the setter for `PreferenceSettings.SelectionPackagePathForInstall` - if `selectionPackagePathForInstall is null or empty string` then it tries to get the UserDataFolder from the PathManager - but in this test harness DynamoModel has not been created and the event is not hooked up.

I couldn't find a nice way to get access to the internal `SourceMigrator.PreferenceSettings` to hookup the event, but maybe I can mock it. I just didn't think it was worth it for 2.12 as opposed to moving onto the next bug.

Let me know.

Reply:
That's fine. LGTM!